awx/ui_next/src/screens/Host/HostGroups/HostGroupsList.jsx

```diff
@@ -48,26 +66,68 @@ function HostGroupsList({ i18n, location, match }) {
     fetchGroups();
   }, [fetchGroups]);
 
-  const handleSelectAll = isSelected => {
-    setSelected(isSelected ? [...groups] : []);
-  };
+  const { selected, isAllSelected, handleSelect, setSelected } = useSelected(
+    groups
+  );
 
-  const handleSelect = row => {
-    if (selected.some(s => s.id === row.id)) {
-      setSelected(selected.filter(s => s.id !== row.id));
-    } else {
-      setSelected(selected.concat(row));
+  const {
+    isLoading: isDisassociateLoading,
+    deleteItems: disassociateHosts,
+    deletionError: disassociateError,
+  } = useDeleteItems(
+    useCallback(async () => {
+      return Promise.all(
+        selected.map(group => HostsAPI.disassociateGroup(hostId, group))
+      );
+    }, [hostId, selected]),
+    {
+      qsConfig: QS_CONFIG,
+      allItemsSelected: isAllSelected,
+      fetchItems: fetchGroups,
     }
+  );
+
+  const handleDisassociate = async () => {
+    await disassociateHosts();
+    setSelected([]);
   };
 
-  const isAllSelected =
-    selected.length > 0 && selected.length === groups.length;
+  const fetchGroupsToAssociate = useCallback(
```

Comment:
Why not use useRequest here?

Reply:
That is because the request is not getting called in this component. On line 95, the function is getting "defined" for later use. `fetchGroupsToAssociate` is passed to AssociateModal as the `fetchRequest` prop. When the modal opens, `fetchRequest` is called within the `useRequest` hook to populate the modal list.